irohad/main/irohad.cpp

```diff
@@ -173,8 +173,12 @@ int main(int argc, char *argv[]) {
 
   // check if at least one block is available in the ledger
   auto blocks_exist = false;
-  irohad.storage->getBlockQuery()->getTopBlocks(1).subscribe(
-      [&blocks_exist](auto block) { blocks_exist = true; });
+  irohad.storage->getBlockQuery()->getTopBlock().match(
+      [&blocks_exist](iroha::expected::Value<
+                      std::shared_ptr<shared_model::interface::Block>> block) {
+        blocks_exist = true;
+      },
+      [](iroha::expected::Error<std::string> error) {});
```

Comment:
Can be simplified to:
<img width="525" alt="image" src="https://user-images.githubusercontent.com/14168731/41532204-c3f8e688-72fe-11e8-8afb-2805eb214ff1.png">